python/libsettings.pyx

```diff
@@ -148,21 +148,23 @@ cdef class Settings:
     def destroy(self):
         settings_destroy(&self.ctx)
 
-    def write(self, section, name, value):
+    def write(self, section, name, value, encoding='ascii'):
+        if not isinstance(value, str):
+            value = str(value).encode(encoding)
         return settings_write_str(self.ctx,
-                                  bytes(section),
-                                  bytes(name),
-                                  bytes(str(value)))
+                                  bytearray(section, encoding),
+                                  bytearray(name, encoding),
+                                  bytearray(value, encoding))
```

Comment:
isn't there a risk of double encoding with this line and `value = str(value).encode(encoding)` above?

Reply:
yes, reworked